contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,582 @@
+    }
+
+    /**
+     * Removes an issuance module from the whitelist
+     *
+     * @param _issuanceModuleAddress    Address of issuance module to remove
+     */
+    function removeIssuanceModule(address _issuanceModuleAddress) public onlyOwner {
+        _removeIssuanceModule(_issuanceModuleAddress);
+    }
+
+    /**
+     * Change the _swapTarget
+     *
+     * @param _swapTarget    Address of the swap target contract. (Usually ZeroEx ExchangeProxy)
+     */
+    function setSwapTarget(address _swapTarget) public onlyOwner {
+        swapTarget = _swapTarget;
+    }
+
+    /**
+     * Runs all the necessary approval functions required for a given ERC20 token.
+     * This function can be called when a new token is added to a SetToken during a
+     * rebalance.
+     *
+     * @param _token    Address of the token which needs approval
+     */
+    function approveToken(IERC20 _token, address _spender) public  isWhitelistedIssuanceModule(_spender) {
+        _safeApprove(_token, _spender, type(uint96).max);
+    }
+
+    /**
+     * Runs all the necessary approval functions required for a list of ERC20 tokens.
+     *
+     * @param _tokens    Addresses of the tokens which need approval
+     */
+    function approveTokens(IERC20[] calldata _tokens, address _spender) external {
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            approveToken(_tokens[i], _spender);
+        }
+    }
+
+    /**
+     * Runs all the necessary approval functions required before issuing
+     * or redeeming a SetToken. This function need to be called only once before the first time
+     * this smart contract is used on any particular SetToken.
+     *
+     * @param _setToken    Address of the SetToken being initialized
+     */
+    function approveSetToken(ISetToken _setToken, address _issuanceModule) isSetToken(_setToken) external {
+        address[] memory components = _setToken.getComponents();
+        for (uint256 i = 0; i < components.length; i++) {
+            // Check that the component does not have external positions
+            require(
+                _setToken.getExternalPositionModules(components[i]).length == 0,
+                "ExchangeIssuance: EXTERNAL_POSITIONS_NOT_ALLOWED"
+            );
+            approveToken(IERC20(components[i]), _issuanceModule);
+        }
+    }
+
+    /**
+    * Issues an exact amount of SetTokens for given amount of input ERC20 tokens.
+    * The excess amount of tokens is returned in an equivalent amount of ether.
+    *
+    * @param _setToken              Address of the SetToken to be issued
+    * @param _inputToken            Address of the input token
+    * @param _amountSetToken        Amount of SetTokens to issue
+    * @param _maxAmountInputToken   Amount of SetTokens to issue
+    * @param _componentQuotes       The encoded 0x transactions to execute 
+    *
+    * @return amountEthReturn       Amount of ether returned to the caller
```

Comment:
Shouldn't this be amountTokenReturn?

Reply:
Correct. Actually I renamed it to `totalInputTokenSold` to be consistent with the variable name returned in the function body